### PR TITLE
Nox

### PR DESCRIPTION
**Pull request by DorianGray**
`nox` is an explicit pure python alternative to `tox`. It does all the same things as tox but instead of an esoteric ini file for configuration it offers a python script, `noxfile.py` for configuring sessions.

Benefits over tox:
- no need for tox-github-actions anymore
- a little faster
- python instead of ini

There are some somewhat unrelated changes here too, let me know if I should break them out or if this is not welcome at all that's fine too! I've been using this setup for personal qtile development for a bit, although the CI integration is brand new.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -7,8 +7,10 @@ include CONTRIBUTING.md
 exclude .coveragerc
 exclude .pylintrc
 exclude tox.ini
+exclude noxfile.py
 exclude requirements.txt
 exclude requirements-dev.txt
+exclude pip/*
```
Comment (tych0) [inline MANIFEST.in:13]: This seems wrong; you're committing files in pip/ in this commit IIUC.

**Comment by tych0** [comment]
I would prefer not to switch away from tox at this time; this does not seem to be any better than tox, just different IMO.

File: MANIFEST.in (hunk 1 above)
Comment (DorianGray) [reply MANIFEST.in:13]: right, but we don't need the requirements files in the dist do we? Same as requirement.txt and requirements-dev.txt in the preceeding lines

**Comment by DorianGray** [comment]
is there anything you'd like to keep? For instance, I can break out analyze from test so we get faster results on tox and I can itemize our requirements files similarly to here.

**Comment by tych0** [comment]
Sure, if there's a way to speed up our tox usage I'm all for it.

I could go either way on the itemization of the requirements files. It makes it harder to say "give me all the -dev requirements", but organization is nice.

**Comment by DorianGray** [comment]
I can make a dev.txt

**Comment by ramnes** [comment]
It looks like nox doesn't see the segfaults as failures. I'm not sure if this is a good idea or not. :)

And did you forget to remove tox.ini?

For the split of requirement files, what I've seen people do so far is call this directory `requirements/`, and rename `requirements.txt` to `qtile.txt`. I feel it's more intuitive than `pip` because I'm used to type `pip install -r req<tab>`.

Also, I would like to split the CI workflow into different workflows, per Python version. While moving away from Travis, we lost the ability to re-run the tests on one specific Python version, and to my understanding this is the only way to achieve it in GitHub Actions. But it may be outside the scope of this PR.

**Comment by DorianGray** [comment]
tox doesn't either if the segfaults occur in the forked process we use to run qtile in and the test passes in the main process. They behave the same way in that regard. It has to do with our tests, not the venv/command manager.

I did not forget, I figured if we switched we'd use both for a time as people familiarized themselves with nox(not that it's complicated like tox, lol)

I can rename things accordingly, no problem!

I'll look into it. It would certainly be cleaner to have the matrix approach work. I'm planning on making a separate PR to split up CI like that. I'm working on making all the changes but the nox related ones in separate PRs so we can choose our own aventure.

**Comment by ramnes** [comment]
> I did not forget, I figured if we switched we'd use both for a time as people familiarized themselves with nox(not that it's complicated like tox, lol)

I would prefer to replace it completely if we decide to go this route. I don't think we want to support both. :)

**Comment by DorianGray** [comment]
Fair enough!

**Comment by DorianGray** [comment]
https://github.com/qtile/qtile/pull/2102 should address your requirements feedback @ramnes

**Comment by DorianGray** [comment]
There, all split up into separate PRs.

**Comment by DorianGray** [comment]
This requires https://github.com/qtile/qtile/pull/2106 and https://github.com/qtile/qtile/pull/2102

**Comment by tych0** [comment]
I don't think we should switch to nox. It looks much less active than tox, and "it is configured in python" is not a convincing argument to me.

**Comment by DorianGray** [comment]
tych0 says "no"

**Pull request closed without merging**